fdbclient/VersionVector.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * VersionVector.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FDBCLIENT_VERSION_VECTOR_H
+#define FDBCLIENT_VERSION_VECTOR_H
+
+#pragma once
+
+#include <unordered_map>
+
+#include "fdbclient/FDBTypes.h"
+
+struct VersionVector {
+	std::unordered_map<Tag, Version> versions;
+	Version maxVersion; // Specifies the max version in this version vector. (Note:
```

Comment:
You don't serialize this which is potentially a problem?

Reply:
I was thinking we could derive this after de-serialization but thinking more about it you could be right that we would need to serialize this (to set this field properly if the "versions" map is empty). Will do that.

Reply:
Done.